pkg/kubelet/prober/worker_test.go

```diff
@@ -474,3 +475,50 @@ func TestStartupProbeDisabledByStarted(t *testing.T) {
 	expectContinue(t, w, w.doProbe(), msg)
 	expectResult(t, w, results.Success, msg)
 }
+
+func TestGetPodLabelName(t *testing.T) {
```

Comment:
@dgrisonnet thx for the continual feedback!

Does the below set of cases prove out everything we want?

Reply:
gentle ping @dgrisonnet @SergeyKanzhelev 

This PR is ready for another "consider for merge" review

Reply:
The new changes looks good from my side, but I would love to have a second pair of eyes on these assumptions: https://github.com/kubernetes/kubernetes/pull/104484#discussion_r788750303